src/test/java/de/tum/in/www1/artemis/TextAssessmentIntegrationTest.java

```diff
@@ -112,6 +119,70 @@ public void testPrepareSubmissionForAssessment() {
         assertThat(result).isPresent();
     }
 
+    @Test
+    @WithMockUser(value = "instructor1", roles = "TA")
+    public void testPrepareSubmissionForAssessmentAutomaticLabel() {
+        // create two text blocks
+        int submissionCount = 2;
+        int submissionSize = 1;
+        int numberOfBlocksTotally = submissionCount * submissionSize;
+        var textBlocks = textExerciseUtilService.generateTextBlocksWithIdenticalTexts(numberOfBlocksTotally);
+
+        // Create Exercise to save blocks & submissions
+        TextExercise textExercise = textExerciseUtilService.createSampleTextExerciseWithSubmissions(course, new ArrayList<>(textBlocks), submissionCount, submissionSize);
+        textExercise.setMaxPoints(1000D);
+        exerciseRepository.save(textExercise);
+
+        // Create a cluster and set minimal distance to < Threshold = 1
+        int[] clusterSizes = { 2 };
+        List<TextCluster> clusters = textExerciseUtilService.addTextBlocksToCluster(textBlocks, clusterSizes, textExercise);
+        double[][] minimalDistanceMatrix = new double[numberOfBlocksTotally][numberOfBlocksTotally];
+        // Fill each row with an arbitrary fixed value < 1 to stimulate a simple case of 10 automatic feedback suggestions
+        for (double[] row : minimalDistanceMatrix) {
+            Arrays.fill(row, 0.1);
+        }
+        clusters.get(0).blocks(textBlocks.stream().toList()).distanceMatrix(minimalDistanceMatrix);
+        textClusterRepository.saveAll(clusters);
+
+        // save textBLocks
+        textBlockRepository.saveAll(textBlocks);
+
+        var listOfSubmissions = textSubmissionRepository.getTextSubmissionsWithTextBlocksByExerciseId(textExercise.getId());
+        TextSubmission firstSubmission = listOfSubmissions.get(0);
+        TextSubmission secondSubmission = listOfSubmissions.get(1);
+
+        // Create and set a sample result for submissions
+        Result firstResult = createSampleResultForSubmission(firstSubmission);
+        Result secondResult = createSampleResultForSubmission(secondSubmission);
+
+        // Create a manual feedback and attach first block to it as reference
+        Feedback feedback = new Feedback().credits(10.0).type(FeedbackType.MANUAL).detailText("gj");
+        feedback.setReference(textBlocks.stream().toList().get(0).getId());
+
+        // Set feedback to result
+        firstResult.setFeedbacks(Collections.singletonList(feedback));
```

Comment:
There is a method called `addFeedbackToResult` in `DatabaseUtilService`. can be used in here...

Reply:
I realized this, but that other method adds it in a different way from how I want, it saves the feedback beforehand instead of afterwards.  I did give it a try anyway, and it actually fails, I think it expects a different kind of result, not an empty one. 

Refactored a bit this part though. used a shorter method.